### PR TITLE
Return 401 instead of 403

### LINKED ISSUE #918: Container registry should raise 401 instead of 403 in case of bad credentials during login
```
$ podman login pulp3-source-fedora34.fluffy.example.com
Username: no-such-user
Password: 
Error: authenticating creds for "pulp3-source-fedora34.fluffy.example.com": Requesting bear token: invalid status code from registry 403 (Forbidden)
(pulp) [vagrant@pulp3-source-fedora34 pulp_container]$ podman login pulp3-source-fedora34.fluffy.example.com
Username: admin ( provide wrong password)                    
Password: 
Error: authenticating creds for "pulp3-source-fedora34.fluffy.example.com": Requesting bear token: invalid status code from registry 403 (Forbidden)

(pulp) [vagrant@pulp3-source-fedora34 pulp_container]$ #disable token_auth in pulp-container
(pulp) [vagrant@pulp3-source-fedora34 pulp_container]$ sudo vi /etc/pulp/settings.py 
(pulp) [vagrant@pulp3-source-fedora34 pulp_container]$ prestart
systemctl restart pulpcore-content pulpcore-worker@1 pulpcore-worker@2 pulpcore-resource-manager pulpcore-api

(pulp) [vagrant@pulp3-source-fedora34 pulp_container]$ podman login pulp3-source-fedora34.fluffy.example.com
Username: admin (with wrong password)
Password: 
Error: error logging into "pulp3-source-fedora34.fluffy.example.com": invalid username/password
(pulp) [vagrant@pulp3-source-fedora34 pulp_container]$ podman login pulp3-source-fedora34.fluffy.example.com
Username: no-such-user
Password: 
Error: error logging into "pulp3-source-fedora34.fluffy.example.com": invalid username/password
(pulp) [vagrant@pulp3-source-fedora34 pulp_container]$ #compare to docker hub
(pulp) [vagrant@pulp3-source-fedora34 pulp_container]$ podman login docker.io
Username: no-such-user
Password: 
Error: error logging into "docker.io": invalid username/password
(pulp) [vagrant@pulp3-source-fedora34 pulp_container]$ podman login docker.io
Username: ipanova ( with wrong password)
Password: 
Error: error logging into "docker.io": invalid username/password
```
API calls:
```
$ http  'https://pulp3-source-fedora34.fluffy.example.com/token/?account=admin&service=pulp3-source-fedora34.fluffy.example.com' --verify=no --auth admin:lala
HTTP/1.1 403 Forbidden
Access-Control-Expose-Headers: Correlation-ID
Allow: GET, HEAD, OPTIONS
Connection: keep-alive
Content-Length: 39
Content-Type: application/json
Correlation-ID: a6c63fb8ac904c388e2952317cf19746
Date: Wed, 13 Jul 2022 11:43:31 GMT
Referrer-Policy: same-origin
Server: nginx/1.20.1
Vary: Accept, Cookie
X-Content-Type-Options: nosniff
X-Frame-Options: DENY

{
    "detail": "Invalid username/password."
}


(pulp) [vagrant@pulp3-source-fedora34 pulp_container]$ http  'https://auth.docker.io/token?account=admin&service=registry.docker.io' --verify=no --auth admin:lala
HTTP/1.1 401 Unauthorized
content-length: 45
content-type: application/json; charset=utf-8
date: Wed, 13 Jul 2022 11:44:01 GMT
strict-transport-security: max-age=31536000
www-authenticate: Basic realm="auth.docker.io"
x-trace-id: 04393aa5c09fe2a42e87130573725830

{
    "details": "incorrect username or password"
}

```
https://github.com/encode/django-rest-framework/blob/master/rest_framework/views.py#L448
https://github.com/pulp/pulp_container/blob/main/pulp_container/app/registry_api.py#L240
https://www.django-rest-framework.org/api-guide/exceptions/#exception-handling-in-rest-framework-views

### PR DESCRIPTION
**Pull request by lubosmj**
closes #918

### CODE REVIEW

**Comment by lubosmj** [comment]
```
(pulp) [vagrant@pulp3-source-fedora35 ~]$ podman login pulp3-source-fedora35.localhost.example.com
Username: asfd
Password: 
Error: error logging into "pulp3-source-fedora35.localhost.example.com": invalid username/password
(pulp) [vagrant@pulp3-source-fedora35 ~]$ podman login docker.io
Username: asfdsadfads
Password: 
Error: error logging into "docker.io": invalid username/password
```

File: pulp_container/app/registry_api.py [hunk 1]
```diff
@@ -355,6 +356,7 @@ class BearerTokenView(APIView):
     """
 
     # Allow everyone to access but still value authenticated users.
+    authentication_classes = [BasicAuthentication]
```
Comment (mdellweg) [inline pulp_container/app/registry_api.py:359]: Does it still allow handing out anonymous tokens?

Comment (lubosmj) [reply pulp_container/app/registry_api.py:359]: Yes:
```
(pulp) [vagrant@pulp3-source-fedora35 ~]$ http -vv https://pulp3-source-fedora35.localhost.example.com/token/?service="localhost:24817"
GET /token/?service=localhost:24817 HTTP/1.1
Accept: */*
Accept-Encoding: gzip, deflate
Connection: keep-alive
Host: pulp3-source-fedora35.localhost.example.com
User-Agent: HTTPie/3.2.1



HTTP/1.1 200 OK
Access-Control-Expose-Headers: Correlation-ID
Allow: GET, HEAD, OPTIONS
Connection: keep-alive
Content-Length: 1077
Content-Type: application/json
Correlation-ID: a82a1399d7c043b498a13efd94fa6af2
Date: Thu, 18 Aug 2022 06:55:40 GMT
Referrer-Policy: same-origin
Server: nginx
Vary: Accept
X-Content-Type-Options: nosniff
X-Frame-Options: DENY

{
    "access_token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NiIsImtpZCI6Ik1aU1Q6U1pERTpIRVpUOkVNQlo6R0ZTRzpDTUJVOk1ZM0Q6Q09CWjpNUlNXOkdOUlQ6R1kyVDpTWlJTIn0.eyJhY2Nlc3MiOltdLCJhdWQiOiJsb2NhbGhvc3Q6MjQ4MTciLCJleHAiOjE2NjA4MDYwNDAsImlhdCI6MTY2MDgwNTc0MCwiaXNzIjoiaHR0cHM6Ly9wdWxwMy1zb3VyY2UtZmVkb3JhMzUubG9jYWxob3N0LmV4YW1wbGUuY29tL3Rva2VuLyIsImp0aSI6ImEyNDFiZTNiLTQ3NTgtNDA5NC1iNTQ4LWJmYzU2ZjAxMjU0MiIsIm5iZiI6MTY2MDgwNTc0MCwic3ViIjoiIn0.3FQ7QLLAzuMEPhbVfocPyyvaiRk9Eifus_NHDWgaB3-NDYBH1n57AIE95mDPhMayK5kKpG41DolG4_Wyvt1azw",
    "expires_in": 300,
    "issued_at": "2022-08-18T06:55:40.471912Z",
    "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NiIsImtpZCI6Ik1aU1Q6U1pERTpIRVpUOkVNQlo6R0ZTRzpDTUJVOk1ZM0Q6Q09CWjpNUlNXOkdOUlQ6R1kyVDpTWlJTIn0.eyJhY2Nlc3MiOltdLCJhdWQiOiJsb2NhbGhvc3Q6MjQ4MTciLCJleHAiOjE2NjA4MDYwNDAsImlhdCI6MTY2MDgwNTc0MCwiaXNzIjoiaHR0cHM6Ly9wdWxwMy1zb3VyY2UtZmVkb3JhMzUubG9jYWxob3N0LmV4YW1wbGUuY29tL3Rva2VuLyIsImp0aSI6ImEyNDFiZTNiLTQ3NTgtNDA5NC1iNTQ4LWJmYzU2ZjAxMjU0MiIsIm5iZiI6MTY2MDgwNTc0MCwic3ViIjoiIn0.3FQ7QLLAzuMEPhbVfocPyyvaiRk9Eifus_NHDWgaB3-NDYBH1n57AIE95mDPhMayK5kKpG41DolG4_Wyvt1azw"
}
```

Comment (lubosmj) [reply pulp_container/app/registry_api.py:359]: Actually, give me a second to test this thing properly.

Comment (lubosmj) [reply pulp_container/app/registry_api.py:359]:
```
(pulp) [vagrant@pulp3-source-fedora35 ~]$ http -vv https://pulp3-source-fedora35.localhost.example.com/token/?service="pulp3-source-fedora35.localhost.example.com"
GET /token/?service=pulp3-source-fedora35.localhost.example.com HTTP/1.1
Accept: */*
Accept-Encoding: gzip, deflate
Connection: keep-alive
Host: pulp3-source-fedora35.localhost.example.com
User-Agent: HTTPie/3.2.1



HTTP/1.1 200 OK
Access-Control-Expose-Headers: Correlation-ID
Allow: GET, HEAD, OPTIONS
Connection: keep-alive
Content-Length: 1151
Content-Type: application/json
Correlation-ID: b251aec2b55a4d06aa45a885f5173a81
Date: Thu, 18 Aug 2022 07:10:17 GMT
Referrer-Policy: same-origin
Server: nginx
Vary: Accept
X-Content-Type-Options: nosniff
X-Frame-Options: DENY

{
    "access_token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NiIsImtpZCI6Ik1aU1Q6U1pERTpIRVpUOkVNQlo6R0ZTRzpDTUJVOk1ZM0Q6Q09CWjpNUlNXOkdOUlQ6R1kyVDpTWlJTIn0.eyJhY2Nlc3MiOltdLCJhdWQiOiJwdWxwMy1zb3VyY2UtZmVkb3JhMzUubG9jYWxob3N0LmV4YW1wbGUuY29tIiwiZXhwIjoxNjYwODA2OTE3LCJpYXQiOjE2NjA4MDY2MTcsImlzcyI6Imh0dHBzOi8vcHVscDMtc291cmNlLWZlZG9yYTM1LmxvY2FsaG9zdC5leGFtcGxlLmNvbS90b2tlbi8iLCJqdGkiOiJhYWQ4M2YwNi01MTM0LTQ1ZDktYjc3ZC00YzhjMzQzMzUwYTMiLCJuYmYiOjE2NjA4MDY2MTcsInN1YiI6IiJ9.19Ap5L2Hs4vFrz2-akOHOZ-WCh4wHO_zeom7M_U3DQNOY5v_k3z5u2Zihtpkodvce4RGbkMlPA0Pi-I5HRs0KQ",
    "expires_in": 300,
    "issued_at": "2022-08-18T07:10:17.004930Z",
    "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NiIsImtpZCI6Ik1aU1Q6U1pERTpIRVpUOkVNQlo6R0ZTRzpDTUJVOk1ZM0Q6Q09CWjpNUlNXOkdOUlQ6R1kyVDpTWlJTIn0.eyJhY2Nlc3MiOltdLCJhdWQiOiJwdWxwMy1zb3VyY2UtZmVkb3JhMzUubG9jYWxob3N0LmV4YW1wbGUuY29tIiwiZXhwIjoxNjYwODA2OTE3LCJpYXQiOjE2NjA4MDY2MTcsImlzcyI6Imh0dHBzOi8vcHVscDMtc291cmNlLWZlZG9yYTM1LmxvY2FsaG9zdC5leGFtcGxlLmNvbS90b2tlbi8iLCJqdGkiOiJhYWQ4M2YwNi01MTM0LTQ1ZDktYjc3ZC00YzhjMzQzMzUwYTMiLCJuYmYiOjE2NjA4MDY2MTcsInN1YiI6IiJ9.19Ap5L2Hs4vFrz2-akOHOZ-WCh4wHO_zeom7M_U3DQNOY5v_k3z5u2Zihtpkodvce4RGbkMlPA0Pi-I5HRs0KQ"
}


Elapsed time: 0.025852835s

(pulp) [vagrant@pulp3-source-fedora35 ~]$ http https://pulp3-source-fedora35.localhost.example.com/v2/ --auth-type=jwt --auth=eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NiIsImtpZCI6Ik1aU1Q6U1pERTpIRVpUOkVNQlo6R0ZTRzpDTUJVOk1ZM0Q6Q09CWjpNUlNXOkdOUlQ6R1kyVDpTWlJTIn0.eyJhY2Nlc3MiOltdLCJhdWQiOiJwdWxwMy1zb3VyY2UtZmVkb3JhMzUubG9jYWxob3N0LmV4YW1wbGUuY29tIiwiZXhwIjoxNjYwODA2OTE3LCJpYXQiOjE2NjA4MDY2MTcsImlzcyI6Imh0dHBzOi8vcHVscDMtc291cmNlLWZlZG9yYTM1LmxvY2FsaG9zdC5leGFtcGxlLmNvbS90b2tlbi8iLCJqdGkiOiJhYWQ4M2YwNi01MTM0LTQ1ZDktYjc3ZC00YzhjMzQzMzUwYTMiLCJuYmYiOjE2NjA4MDY2MTcsInN1YiI6IiJ9.19Ap5L2Hs4vFrz2-akOHOZ-WCh4wHO_zeom7M_U3DQNOY5v_k3z5u2Zihtpkodvce4RGbkMlPA0Pi-I5HRs0KQ
HTTP/1.1 200 OK
Access-Control-Expose-Headers: Correlation-ID
Allow: GET, HEAD, OPTIONS
Connection: keep-alive
Content-Length: 2
Content-Type: application/json
Correlation-ID: 36537c439271499d887520dfda972206
Date: Thu, 18 Aug 2022 07:10:26 GMT
Docker-Distribution-Api-Version: registry/2.0
Referrer-Policy: same-origin
Server: nginx
Vary: Accept
X-Content-Type-Options: nosniff
X-Frame-Options: DENY
X-Registry-Supports-Signatures: 1

{}
```

**Review by mdellweg — APPROVED** [review]

**Review by ipanova — APPROVED** [review]
I did some manual testing and it works :+1:

**Pull request merged**